src/averageAges.js

```diff
@@ -37,7 +48,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filterArr = people.filter((person) => {
+    const fullName = person.name;
+
+    const hasChildren = (human) => fullName === human.mother;
+
+    return (person.sex === 'f' && people.some(hasChildren))
+      || (person.sex === 'f' && (!withChildren));
```

Comment:
you can do it like you did in calculateMenAverageAge, if hasChildren true then you do all of this, and you can use your helper function to find all women